### PR TITLE
feat-4: Reusable components

### PR DESCRIPTION
**Pull request by JediDevYury**
#4

### CODE REVIEW

File: components/ui/button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+      "3xl": "gap-7",
+      "4xl": "gap-8",
+    },
+    isAttached: {
+      true: "gap-0",
+    },
+  },
+});
+
+type IButtonProps = Omit<
```
Comment (angelinabylitskaya) [inline components/ui/button/index.tsx:291]: Styles can be moved to a separate file to make current file smaller and readable

File: components/ui/input/index.tsx [hunk 2]
```diff
@@ -0,0 +1,290 @@
+"use client";
```
Comment (angelinabylitskaya) [inline components/ui/input/index.tsx]: is this line required?

File: providers/useThemeProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+"use client";
```
Comment (angelinabylitskaya) [inline providers/useThemeProvider.tsx]: is this line required?

File: components/ui/button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,491 @@
+"use client";
```
Comment (MZubrytski) [inline components/ui/button/index.tsx]: the same

**Review by MZubrytski — APPROVED** [review]

**Pull request merged**